src/lucky/renderable.cr

```diff
@@ -103,19 +103,66 @@ module Lucky::Renderable
     response.print
   end
 
+  private def handle_response(_response : Nil) forall T
+    # This is a hack to force Crystal to regenerate this method for
+    # each action. That way @type refers to the action with the problem,
+    # and not the first action to be compiled.
+    {% @type %}
+    {%
+      raise <<-ERROR
+
+
+      #{@type} returned Nil
+
+      But it should return a Lucky::Response.
+
+      Try this...
+
+        ▸ Ensure the action returns a response with html/redirect/json/etc.
+        ▸ Ensure all conditionals (like if/else) return a response with html/redirect/json/etc.
+
+      For example...
+
+        get "/admin/users" do
+          # Make sure there is a response in all conditional branches
+          if current_user.admin?
+            html IndexPage, users: UserQuery.new
+          else
+            redirect Home::Index
+          end
+        end
+
+      ERROR
+    %}
+  end
+
   private def handle_response(_response : T) forall T
+    # This is a hack to force Crystal to regenerate this method for
+    # each action. That way @type refers to the action with the problem,
+    # and not the first action to be compiled.
+    {% @type %}
     {%
       raise <<-ERROR
 
 
-      Your action returned #{T}
+      #{@type} returned #{T}
 
       But it should return a Lucky::Response
 
       Try this...
 
-        ▸ Use a method like render/redirect/json at the end of your action.
-        ▸ Ensure all conditionals (like if/else) return a response with render/redirect/json/etc.
+        ▸ Use a method like html/redirect/json at the end of your action.
+        ▸ Ensure all conditionals (like if/else) return a response with html/redirect/json/etc.
+
+      For example...
+
+        get "/users" do
+          # Render html, json, or redirect
+          html IndexPage, users: UserQuery.new
+          redirect to: Home::Index
+          json({name: "James"})
+        end
```

Comment:
Added an example here too

Reply:
I liked your realistic example above instead of showing all possible options. I think showing a different example could be good, but I think it should still work if it's an example. 🙂

Reply:
Yeah good point!